src/config/routes.js

```diff
@@ -13,18 +14,24 @@ import Geneology from '../pages/Geneology/Geneology';
 import GetInvolved from '../pages/GetInvolved/GetInvolved';
 
 const Routes = () => {
-  return (
-    <Switch>
-      <Route exact path="/" component={Home} />
-      <Route path="/about" component={About} />
-      <Route path="/events" component={Events} />
-      <Route path="/join" component={GetInvolved} />
-      <Route path="/news" component={News} />
-      <Route path="/contact" component={Contact} />
-      <Route path="/donate" component={Donate} />
-      <Route path="/ogsh" component={Geneology} />
-    </Switch>
-  );
+  const pageState = useSelector(state => state.pages);
+
+  if (pageState) {
+    return (
+      <Switch>
+        <Route exact path="/" render={(props) => <Home {...props} data={pageState.home} />} />
+        <Route path="/about" component={About} />
+        <Route path="/events" component={Events} />
+        <Route path="/join" component={GetInvolved} />
+        <Route path="/news" component={News} />
+        <Route path="/contact" component={Contact} />
+        <Route path="/donate" component={Donate} />
+        <Route path="/ogsh" component={Geneology} />
+      </Switch>
+    );
+  }
+
+  return null;
```

Comment:
🤔 putting the routes in a conditional wrapper like this feels wrong instead should we just grab data from the store per page? its not a network call so it should be rather fast

Reply:
Currently, { state.pages } returns all the page's ACF data. Looking for page data before state.pages will through an undefined error. i.e. { state.pages.home  == undefined }. 

We can still do it at page level but it would be the same conditional ({  if (state.pages) }) and would be a bit redundant. LMK what you think.

Reply:
correct `state.pages.home` is super nested but you **can** get a solution without wrapping in a conditional

Reply:
I'm fine with it not being page level, im not fine with the conditional logic, you can use redux to a bit different to get an outcome where you dont need a conditional render